src/halmos/__main__.py

```diff
@@ -408,7 +408,7 @@ def run(
             #
             log=deepcopy(setup_ex.log),
             cnts=deepcopy(setup_ex.cnts),
-            sha3s=deepcopy(setup_ex.sha3s),
+            sha3s=deepcopy(setup_ex.sha3s),  # TODO: shallow copy
```

Comment:
wouldn't it make more sense to actually not copy sha3s at all and just hang on to the reference?

Reply:
you mean reusing the sha3s dict for different tests?  no it doesn't work, because it will skip the no hash collision assumption for the existing hash expressions in the dict.